routing/single_mwm_router.cpp

```diff
@@ -137,17 +142,23 @@ IRouter::ResultCode SingleMwmRouter::DoCalculateRoute(MwmSet::MwmId const & mwmI
   case AStarAlgorithm<IndexGraphStarter>::Result::NoPath: return IRouter::RouteNotFound;
   case AStarAlgorithm<IndexGraphStarter>::Result::Cancelled: return IRouter::Cancelled;
   case AStarAlgorithm<IndexGraphStarter>::Result::OK:
-    if (!RedressRoute(mwmId, *vehicleModel, routingResult.path, delegate, starter, route))
+    if (!RedressRoute(routingResult.path, delegate, starter, route))
       return IRouter::InternalError;
     if (delegate.IsCancelled())
       return IRouter::Cancelled;
     return IRouter::NoError;
   }
 }
 
-bool SingleMwmRouter::FindClosestEdge(MwmSet::MwmId const & mwmId, m2::PointD const & point,
+bool SingleMwmRouter::FindClosestEdge(platform::CountryFile const & file, m2::PointD const & point,
                                       Edge & closestEdge) const
 {
+  MwmSet::MwmHandle handle = m_index.GetMwmHandleByCountryFile(file);
```

Comment:
The mwm is the same for each FindClosestEdge() call, why not to take handle once, before call to FindClosestEdge?